test/common.py

```diff
@@ -686,6 +686,10 @@ def setUp(self):
     self.wasm_engines = config.WASM_ENGINES.copy()
     self.banned_js_engines = []
     self.use_all_engines = EMTEST_ALL_ENGINES
+    if self.js_engines[0] != config.NODE_JS:
+      # If our primary JS engine is something other than node then enable
+      # shell support
+      self.set_setting('ENVIRONMENT', 'web,node,shell')
```

Comment:
We probably want `worker` here as well. But, how about returning to the default? The empty string does that:
```suggestion
      self.set_setting('ENVIRONMENT', '')
```

Reply:
`shell` is not in the default so this needs to be whatever the default set is plus `shell`.

Reply:
Remember this only effects folks how put a non-node engine first in JS_ENGINES list.. which I imagine is basically just our CI systems.